service/controller/rolebindingtemplate/resource/rolebinding/resource.go

```diff
@@ -62,44 +62,70 @@ func (r *Resource) Name() string {
 	return Name
 }
 
-func (r *Resource) getNamespacesFromScope(ctx context.Context, scopes v1alpha1.RoleBindingTemplateScopes) ([]string, error) {
+func getLabelSelectorFromScopes(scopes v1alpha1.RoleBindingTemplateScopes) (labels.Selector, error) {
```

Comment:
functionally, this is the only change. Everything below is just moving existing logic into separate functions to make this easier to read/test